core/serialization/deserializer/montage-reviver.js

```diff
@@ -341,7 +337,7 @@ var MontageReviver = exports.MontageReviver = Montage.specialize(/** @lends Mont
             // should be rejected as an error.
             error = this._checkLabel(label, isAlias);
             if (error) {
-                return Promise.reject(error);
+                throw error;
```

Comment:
shouldn't there be test if not sync then Promise.reject(error) as before?

Reply:
It's not strictly necessary because the montage-deserializer does a try/catch over context API calls to catch synchronous errors as well as rejected promises. Since nothing but the deserializer uses context APIs, why create an extra promise when we could just throw a regular error?